File: server/utils/encryption.go
package utils

import (
	"crypto/aes"
	"crypto/cipher"
	"crypto/rand"
)

func Encrypt(message string, secretKey string) (string, error) {
	aes, err := aes.NewCipher([]byte(secretKey))
	if err != nil {
		panic(err)
	}

	gcm, err := cipher.NewGCM(aes)
	if err != nil {
		panic(err)
	}


	nonce := make([]byte, gcm.NonceSize())
	_, err = rand.Read(nonce)
	if err != nil {
		panic(err)
	}

	ciphertext := gcm.Seal(nonce, nonce, []byte(message), nil)

	return string(ciphertext),nil
}

func Decrypt(encryptedMessage string, secretKey string) (string, error) {
	aes, err := aes.NewCipher([]byte(secretKey))
	if err != nil {
		panic(err)
	}

	gcm, err := cipher.NewGCM(aes)
	if err != nil {
		panic(err)
	}

	
	nonceSize := gcm.NonceSize()
	nonce, encryptedMessage := encryptedMessage[:nonceSize], encryptedMessage[nonceSize:]

	plaintext, err := gcm.Open(nil, []byte(nonce), []byte(encryptedMessage), nil)
	if err != nil {
		panic(err)
	}

	return string(plaintext),nil
}

File: server/src/utils/encryption.ts
import CryptoJS from 'crypto-js'

export const encrypt = (message: string, secretKey: string): string => {
  const encryptedMessage = CryptoJS.AES.encrypt(message, secretKey).toString()
  return encryptedMessage
}

export const decrypt = (message: string, secretKey: string): string => {
  const decryptedBytes = CryptoJS.AES.decrypt(message, secretKey)
  const decryptedMessage = decryptedBytes.toString(CryptoJS.enc.Utf8)
  return decryptedMessage
}
